### ISSUE TITLE
[no-unsafe-assignment] unable to cast in `.js` files

### ISSUE THREAD

**Issue by DetachHead** [typescript-eslint/typescript-eslint#4824]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

eslintrc:
```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-assignment": "error"
  }
}
```

code:
```js
const foo = /** @type {string} */ (require('foo'))
```

tsconfig.json:
```json
{
    "compilerOptions": { "allowJs": true, "checkJs": true },
    "include": ["**/*.js"]
}
```

**Expected Result**

no error, as this is a cast which is allowed in typescript files
```ts
const foo = require('asdf') as string // allowed
const bar: string = require('asdf') // error
```

**Actual Result**
```js
//error: Unsafe assignment of an `any` value.
const foo = /** @type {string} */ (require('foo'))
```

**Additional Info**
according to [the docs](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#casts), this syntax is equivalent to a cast, not a type annotation. therefore `no-unsafe-assignment` should allow it, to be consistent with how it works with `.ts` files

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.18.0` |
| `@typescript-eslint/parser`        | `5.18.0` |
| `TypeScript`                       | `4.7.0-beta` |
| `ESLint`                           | `8.13.0` |
| `node`                             | `16.9.1` |

**Comment by bradzacher** [comment]
This is because from an AST perspective this is just an assignment.

See https://github.com/typescript-eslint/typescript-eslint/issues/1682
